### PR TITLE
Encryption config settings.

### PR DESCRIPTION
**Pull request by realgaurav**
## Description

Encryption configuration settings.

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit 243a30f3ac63c96c042995390cd191c54fc992d9

https://deploy-preview-3505--calico.netlify.app

File: reference/resources/felixconfig.md [hunk 1]
```diff
@@ -92,6 +92,11 @@ spec:
 | vxlanMTU                           | MTU to use for the VXLAN tunnel device. Also controls NodePort MTU when eBPF enabled. | int | int | `1410` |
 | vxlanPort                          | Port to use for VXLAN traffic. A value of `0` means "use the kernel default". | int | int | `4789` |
 | vxlanVNI                           | Virtual network ID to use for VXLAN traffic. A value of `0` means "use the kernel default". | int | int | `4096` |
+| wireguardEnabled                   | Enable encryption on WireGuard supported nodes in cluster. When enabled, pod to pod traffic will be sent over encrypted tunnels between the nodes. | `true`, `false` | boolean | `false` |
+| wireguardListeningPort             | Port used by WireGuard tunnels. Felix sets up WireGuard tunnel on each node specified by this port. Please note that this can only be setup in global `FelixConfiguration` resource. Setting it per host, config-file or environment variable will not work. | int | int | 51820 |
```
Comment (bmckercher123) [inline reference/resources/felixconfig.md:96]: Port used by WireGuard tunnels. Felix sets up WireGuard tunnel on each node specified by this port. Available for configuration only in the global `FelixConfiguration` resource; setting it per host, config-file or environment variable will not work.

File: reference/resources/felixconfig.md [hunk 2]
```diff
@@ -92,6 +92,11 @@ spec:
...
+| wireguardRoutingRulePriority       | WireGuard routing rule priority value set up by Felix. Change this to appropriate value if the default value can have an impact on existing routing rules on the node. | int | int | 99 |
```
Comment (bmckercher123) [inline reference/resources/felixconfig.md:97]: This sentence is not clear...are you saying user may need to change the value if the default doesn't work, or is this a warning that changing the default will impact existing routing rules on the node? 

WireGuard routing rule priority value set up by Felix. Change this to appropriate value if the default value can have an impact on existing routing rules on the node.

File: reference/resources/felixconfig.md [hunk 3]
```diff
@@ -92,6 +92,11 @@ spec:
...
+| wireguardInterfaceName             | Name of the WireGuard interface created by Felix. If the value is changed, Felix will not delete earlier configured interface and must be manually deleted on each node, if needed. | string | string | wireguard.cali |
```
Comment (bmckercher123) [inline reference/resources/felixconfig.md:98]: Name of the WireGuard interface created by Felix. If you change the name, and want to clean up the previously-configured interface names on each node, this is a manual process.

**Pull request merged**